Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Paged/RequestBlockingScript.js

```diff
@@ -74,7 +73,7 @@ window.__firefox__.execute(function($) {
     try {
       // We do this in a try/catch block to not fail the request in case we can't
       // create a URL
-      resourceURL = new URL(this._url, window.location.href)
+      resourceURL = new URL(this[localURLProp], window.location.href)
```

Comment:
> reported by [reviewdog](https://github.com/reviewdog/reviewdog) 🐶
> [semgrep] Are you using the `URL(url, base)` constructor as a security control to limit the origin with base `window.location.href`? The base is ignored whenever url looks like an absolute URL, e.g. when it begins `protocol://`. `\\\\` or `//x.y`. Verify that the URL's origin is as expected rather than relying on the URL constructor.
> 
> Source: https://github.com/brave/security-action/blob/main/assets/semgrep_rules/web/url-constructor-base.yaml
> 
> Cc @brave/sec-team @stoletheminerals @thypon

I want it to exactly do that, add the host protocol and port (if available) from the window location

Reply:
Cc @bcaller

Reply:
The logic of this code hasn't changed, it just picked this up as it's a new check.

I don't know what this code is doing so I'm not sure what sort of value `arguments[1] == this[localUrlProp]` is here.

This warning message is about misusing `new URL` to enforce the origin of the URL. The correct use is setting a fallback in case `this[localUrlProp]` is a relative URL. Could it be? Could it be `//a.b.c`? If `resourceURL` ended up on a different origin than the current page would it matter?

Reply:
OK we override all uses of `XMLHttpRequest.prototype.open` in any webpage. Pages can anyway do `new XMLHttpRequest().open('GET', 'http://somewhere.else:1234')` and this code isn't expected to change the host or rewrite anything. Cross origin requests are not supposed to be blocked and this `new URL` is not a security control. It is simply in case the webpage makes a relative request, then we need to set the host, protocol, etc.

Therefore, you can ignore this warning.

I did think there was value in this check as I've seen the URL constructor misused before, but perhaps the wording needs clarification.

Reply:
Is this clearer, or still confusing:

> Are you misusing the URL constructor as a security control to limit the origin to base `window.location.href`? The base is ignored when the URL appears absolute, e.g. beginning `protocol://`, `\\\\` or `//x.y`. Verify the URL's origin rather than relying on the URL ctor. If correctly using this purely as a fallback in case the URL is relative, ignore this.

Reply:
The check is useful but in my case I'm not using the url to make any api calls. I simply construct it as it would be for the original api call and pass it to iOS to determine if this request (with its expected origin) should be blocked or not using the rust AdblockEngine. I expect `this[localURLProp]` (which is essentially `argument[0]` on the `open` function) to be a mix bag of relative paths and full urls. 

We seem to agree this is a valid use case here. Could we then resolve this conversation?

Reply:
Yes. Remove the 'needs-security-review' label (I can't do it myself on this repo).